src/indra_cogex/sources/indra_db/__init__.py

```diff
@@ -107,12 +110,39 @@ def get_relations(self, max_complex_members: int = 3):  # noqa:D102
         logger.info("Loading belief scores per hash")
         with self.belief_scores_fname.open("rb") as f:
             belief_scores = pickle.load(f)
+        logger.info("Loading stmt hash - retraction boolean mapping")
+        has_retracted_pmid = {}
+        with gzip.open(self.stmts_fname, "rt") as fh:
+            reader = csv.reader(fh, delimiter="\t")
+            for sh_str, stmt_json_str in tqdm(
+                    reader, desc="Reading statements", unit_scale=True
+            ):
+                stmt_hash = int(sh_str)
+                stmt_json = load_stmt_json_str(stmt_json_str)
+                evidence = stmt_json["evidence"][0]
+
+                # Get pmid
+                pmid = evidence.get("text_refs", {}).get("PMID") or \
+                    evidence.get("pmid")
+                # If we have a pmid, check if it is retracted.
+                # If this is the first time we see this statement hash,
+                # set the has_retracted_pmid flag to the retraction status,
+                # otherwise, only change it if the pmid is retracted.
+                if pmid is not None:
+                    if stmt_hash not in has_retracted_pmid:
+                        has_retracted_pmid[stmt_hash] = is_retracted(pmid)
+                    else:
+                        # Run OR on the current value and the new value
+                        has_retracted_pmid[stmt_hash] |= is_retracted(pmid)
```

Comment:
are you sure this does the appropriate "or" operation?

Reply:
This convinces me:
```
In [6]: True | False
Out[6]: True

In [7]: myb = False

In [8]: myb |= True

In [9]: myb
Out[9]: True

...
In [19]: myd = {123456: False}

In [20]: myd[123456] |= True

In [21]: myd
Out[21]: {123456: True}
```